File: GOF/Bridge/java/src/Window/IconWindow.java
package Window;

import Elements.Coord;
import Elements.View;
import Window.WindowImp.WindowImp;

public class IconWindow extends Window {

	private String m_bitmapName;

	public IconWindow(View content) {
		super(content);
	}

	@Override
	public void drawContents() {
		WindowImp imp = getWindowImp();
		if(imp != null)
			imp.deviceBitmap(m_bitmapName, new Coord(0), new Coord(0));
	}
}

File: GOF/Bridge/cpp/Window/IconWindow.cpp
//
// Created by 17271 on 2019/3/30.
//

#include "IconWindow.h"
#include "WindowImp/WindowImp.h"
#include <iostream>

using std::cout;			using std::endl;
using std::weak_ptr;

IconWindow::IconWindow(std::weak_ptr<View> contents)
	: Window(contents)
{}

void IconWindow::drawContents()
{
	weak_ptr<WindowImp> imp = getWindowImp();
	if(imp.lock() != nullptr)
		imp.lock()->deviceBitmap(m_bitmapName, 0, 0);
}